### ISSUE TITLE
Template variables parsing

### ISSUE THREAD

**Issue by cusma** [algorandfoundation/algokit-utils-py#56]
Template variables containing `_` are not parsed correctly, raising `DeploymentFailedError` in

```python
def check_template_variables(approval_program: str, template_values: TemplateValueDict) -> None:
   ...
```

**EXAMPLE**

Template variable: `TMPL_ASSET_ID` causes:

`algokit_utils.deploy.DeploymentFailedError: The following template values were not provided: TMPL_ASSET`

**Comment by daniel-makerx** [comment]
Hi @cusma, thanks for reporting this. Will have a fix out for this soon

**Comment by daniel-makerx** [comment]
Released algokit-utils-py 2.1.1 to address this issue
